src/vue/forms/AssetForm.vue

```diff
@@ -261,30 +346,59 @@ export default {
       return this.request instanceof AssetUpdateRequestRecord
     },
   },
-  created () {
-    this.tryPopulateForm(this.request)
+  async created () {
+    await this.tryPopulateForm(this.request)
+    this.isLoaded = true
   },
   methods: {
-    tryPopulateForm (request) {
+    async tryPopulateForm (request) {
       if (request.id) {
-        this.form = {
-          information: {
-            name: request.assetName,
-            code: request.assetCode,
-            maxIssuanceAmount: request.maxIssuanceAmount,
-            logo: request.logo.key
-              ? new DocumentContainer(request.logo)
-              : null,
-            policies: request.policies,
-          },
-          terms: {
-            terms: request.terms.key
-              ? new DocumentContainer(request.terms)
-              : null,
-          },
+        try {
+          const assetCreationRequest =
+            await this.fetchAssetCreationRequest(request.assetCode)
```

Comment:
Is not it duplicates the request from `AssetList.vue` of `fetchAssetUpdateRequest` (line 123)?

Reply:
No it's quite different logic. `fetchAssetUpdateRequests` gets the latest update asset request having pending state (it means that it's processing now), while `fetchAssetCreationRequest` gets the only creation asset request, cause only that request contains all the information about asset pre-issuance (initial preissued amount & asset signer). 

The requests with `assetUpdate` type don't have that kind of data, it's the only right way to get it IMO. That's why we need to have two methods with similar actions, but different purposes.

Reply:
ok, let’s see what @quinn71 will say

Reply:
https://github.com/tokend/web-client/pull/37#discussion_r254282691